contracts/Lens/VenusLens.sol

```diff
@@ -551,6 +577,72 @@ contract VenusLens is ExponentialNoError {
         return rewardSummary;
     }
 
+    /**
+     * @notice Returns all pools (excluding the Core Pool) along with their associated market data
+     * @param comptroller The Comptroller contract to query
+     * @return poolsData An array of PoolWithMarkets structs, each containing pool info and its markets
+     */
+    function getAllPoolsData(
+        ComptrollerInterface comptroller
+    ) external view returns (PoolWithMarkets[] memory poolsData) {
+        uint96 lastPoolId = comptroller.lastPoolId();
+        poolsData = new PoolWithMarkets[](lastPoolId);
+
+        for (uint96 i = 1; i <= lastPoolId; ++i) {
```

Comment:
Should it really exclude the Core Pool ( `poolId`  0) here?

Maybe my understanding is wrong, but I believe we'd have to surface the `MarketData` corresponding to the whole Core Pool, so that we know the values for `liquidationThreshold`, `liquidationIncentive` and `isBorrowAllowed`. Unless there's another way to get this info.